tests/webfinger/server/4__3_only_returns_jrd_in_response_to_https_requests.py

```diff
@@ -41,3 +43,23 @@ def does_not_return_jrd_in_response_to_http(
             f'Returns JRD content.\nAccessed URI: "{ http_webfinger_uri }".',
             spec_level=SpecLevel.MUST,
             interop_level=InteropLevel.UNAFFECTED) # if a server also responds to http, nothing bad happens
+
+
+@test
+def uses_valid_https_certificate(client: WebFingerClient, server: WebFingerServer):
+    account_id = server.obtain_account_identifier()
+    webfinger_request = client.construct_webfinger_uri_for(account_id)
+    try:
+        client.http_get(webfinger_request, verify=True)
+    except httpx.ConnectError as ex:
+        # Unpack the cause of the exception
+        cause = ex
+        while cause := getattr(cause, "__cause__", None):
```

Comment:
Are there some httpx.ConnectErrors that aren't caused by SSLCertVerificationErrors?

Reply:
According to ChatGPT :-) ...

The `httpx.HTTPConnect` exception in the `httpx` library can be caused by a variety of conditions, typically related to issues in establishing a connection to the server. Here are some common conditions that can trigger this exception:

1. **DNS Resolution Failure**: When the client's DNS resolver cannot resolve the hostname to an IP address.
   
2. **Connection Timeout**: When the connection attempt to the server exceeds the configured timeout limit.

3. **Network Unreachability**: When the network is down or the target server is not reachable due to network issues.

4. **Refused Connection**: When the server actively refuses the connection request, often because the server is not listening on the specified port.

5. **Invalid Host**: When the hostname provided is invalid or malformed.

6. **SSL/TLS Handshake Failure**: When the connection requires SSL/TLS and the handshake fails due to certificate issues, protocol mismatches, or other SSL-related problems.

7. **Proxy Issues**: When using a proxy, if the proxy server is unreachable or misconfigured.

8. **Firewall or Security Settings**: When firewall rules or other security settings block the connection attempt.

Understanding the exact cause often requires inspecting the exception message and the context in which the exception was raised.

Reply:
My point being that it might not report on those other potential problems?

Reply:
The intent of this test is to only report on TLS certificate issues.

Reply:
Oh, are you saying we should re-raise the exception to result in an "error" (vs a "failure")? If so, I can add that. I had that in there originally and I think it may have accidentally been removed.

Reply:
So if there is a typo and DNS doesn't resolve, or refused connection or such, that shouldn't be reported in this test? Not as a test failure but as a problem...

Reply:
I added the re-raise.